src/storage/transaction/ChainAddEdgesProcessorLocal.cpp

```diff
@@ -0,0 +1,357 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+
+#include <thrift/lib/cpp/util/EnumUtils.h>
+#include "storage/transaction/ChainAddEdgesProcessorLocal.h"
+#include "storage/transaction/ConsistUtil.h"
+#include "storage/transaction/TransactionManager.h"
+#include "storage/mutate/AddEdgesProcessor.h"
+#include "storage/StorageFlags.h"
+
+namespace nebula {
+namespace storage {
+
+using CODE = ::nebula::cpp2::ErrorCode;
+using SVEC = std::vector<std::string>;
+
+void ChainAddEdgesProcessorLocal::process(const cpp2::AddEdgesRequest& req) {
+    processInternal(req, ChainProcessType::NORMAL);
+}
+
+void ChainAddEdgesProcessorLocal::resumeChain(GraphSpaceID spaceId,
+                                              PartitionID partId,
+                                              folly::StringPiece val) {
+    CHECK(resumeRequest(spaceId, partId, val));
+    std::vector<std::string> keys = strEdgekeys(req_);
+    edgeVers_ = ConsistUtil::getMultiEdgeVers(env_->kvstore_, spaceId, partId, keys);
+    processInternal(req_, ChainProcessType::RESUME_CHAIN);
+}
+
+void ChainAddEdgesProcessorLocal::resumeRemote(GraphSpaceID spaceId,
+                                               PartitionID partId,
+                                               folly::StringPiece val) {
+    CHECK(resumeRequest(spaceId, partId, val));
+    std::vector<std::string> keys = strEdgekeys(req_);
+    edgeVers_ = ConsistUtil::getMultiEdgeVers(env_->kvstore_, spaceId, partId, keys);
+    processInternal(req_, ChainProcessType::RESUME_REMOTE);
+}
+
+void ChainAddEdgesProcessorLocal::processInternal(const cpp2::AddEdgesRequest& req,
+                                                  ChainProcessType type) {
+    req_ = req;
+    processType_ = type;
+    localPartId_ = req.get_parts().begin()->first;
+    auto spaceId = req.get_space_id();
+    const auto& partEdges = req.get_parts();
+    auto ret = env_->schemaMan_->getSpaceVidLen(spaceId);
+    if (!ret.ok()) {
+        LOG(ERROR) << ret.status();
+        for (auto& part : partEdges) {
+            pushResultCode(nebula::cpp2::ErrorCode::E_INVALID_SPACEVIDLEN, part.first);
+        }
+        onFinished();
+        return;
+    }
+
+    spaceVidLen_ = ret.value();
+
+    if (type != ChainProcessType::NORMAL) {
+        std::vector<std::string> keys = strEdgekeys(req);
+        edgeVers_ = ConsistUtil::getMultiEdgeVers(env_->kvstore_, spaceId, localPartId_, keys);
```

Comment:
Where do we use `edgeVers_`? Because I see `ChainAddEdgesProcessorRemote::checkVersion` is checking this version to prevent overwrite newer data.